src/api/app/views/webui/request/_add_role_request_dialog.html.haml

```diff
@@ -18,9 +18,24 @@
           .form-group
             = label_tag(:role, 'Add role:')
             = select_tag(:role, options_for_select([['Maintainer', 'maintainer'], ['Bugowner', 'bugowner']]), class: 'form-control')
-          .form-group
-            = label_tag(:user, 'For user:')
-            = text_field_tag(:user, User.session!.login, class: 'form-control')
+          .card
+            .bg-light
+              %ul.nav.nav-tabs.pt-2.px-3.flex-nowrap.collapsible{ 'role': 'tablist' }
+                %li.nav-item
+                  %a{ 'id' => 'user-tab', 'data-toggle' => 'tab', 'href' => '#user', 'class' => 'nav-link text-nowrap active',
+                   'role' => 'tab', 'aria-controls' => 'user', 'aria-selected' => 'true', 'onclick' => 'deleteGroupInput()' }
+                    For user
+                %li.nav-item
+                  %a{ 'id' => 'group-tab', 'data-toggle' => 'tab', 'href' => '#group', 'class' => 'nav-link text-nowrap',
+                   'role' => 'tab', 'aria-controls' => 'group', 'aria-selected' => 'false', 'onclick' => 'deleteUserInput()' }
+                    For groups
+              %div{ 'class' => 'tab-content' }
+                %div{ 'id' => 'user', 'class' => 'tab-pane fade show active', 'role' => 'tab', 'aria-labelledby' => 'user-tab' }
+                  = text_field_tag(:user, nil, id: 'user-text-field', class: 'form-control', placeholder: 'Enter username',
+                  onclick: 'deleteGroupInput()')
+                %div{ 'id' => 'group', 'class' => 'tab-pane fade', 'role' => 'tab', 'aria-labelledby' => 'group-tab' }
```

Comment:
This is really not a big deal and I don't know if you're new to HAML or not, but just in case you are, the `id` and `class` properties can be written directly after the tag instead of being passed in the attributes (so inside the curly brackets). This would result in:

`%div.tab-pane.fade#group{ 'role' => 'tab', ... }`.

One more trick, the `div` is so commonly used that it can be skipped! So it results in:

`.tab-pane.fade#group{ 'role' => 'tab', ... }`

Voilà!

Reply:
Thanks for the useful tips. I'm fairly new to HAML :)